addons/misra.py

```diff
@@ -740,8 +762,38 @@ def getForLoopCounterVariables(forToken):
                     vars_modified.add(tn.variable)
                 elif tn.previous and tn.previous.str in ('++', '--'):
                     vars_modified.add(tn.variable)
-        if cur_clause == 1 and tn.isAssignmentOp and tn.astOperand1.variable:
-            vars_initialized.add(tn.astOperand1.variable)
+        if cur_clause == 1 and tn.isAssignmentOp:
+            var_token = tn.astOperand1
+            while var_token and var_token.str == '.':
+                var_token = var_token.astOperand2
+            if var_token and var_token.variable:
+                vars_initialized.add(var_token.variable)
+        if cur_clause == 1 and tn.isName and tn.next.str == '(':
+            function_args_in_init = getArguments(tn.next);
+            function_args = {}
+            if tn.function:
+                function_args = tn.function.argument
+            if tn.function is None:
+                for arg_tok in function_args_in_init:
+                    if arg_tok.variable:
+                        vars_initialized.add(arg_tok.variable)
```

Comment:
it is not likely that this variable x is changed here in this expression:  `dostuff(x)`. I suggest that we only add the variable if address-of operator is used.